packages/blade/src/components/Tour/_decisions/decisions.md

```diff
@@ -0,0 +1,766 @@
+
+      <Card>
+        <CardBody>
+          <TourStep name="step-3">
+            <Box>
+              <Text>Some content</Text>
+              <Text>Some content</Text>
+              <Text>Some content</Text>
+            </Box>
+          </TourStep>
+        </CardBody>
+      </Card>
+    </Box>
+  );
+};
+```
+
+## Motion
+
+Check/Provide feedback for the motion for Tour component [here](https://razorpay.slack.com/archives/C0274H7QRC1/p1697520879941439)
+
+https://github.com/razorpay/blade/assets/35374649/5830c059-fbd6-461e-915c-e3a98e930735
+
+---
+
+---
+
+> NOTE: The below sections are discussions, decisions & challenges we faced while designing the API for the `Tour` component.
+> If you are interested in diving deeper into the rabbit hole, you can read the below sections, or else you can skip them.
+
+## Open Questions And Technical Challenges
+
+## React Native Specifics
+
+On react-native there is 2 differences in the API compared to web:
+
+1. `TourStep` component is needed.
+2. `Tour` component needs to wrap the whole app
+
+**1. `TourStep` Component is needed:**
+
+As highlighted earlier in the [API](#api) section, the `TourStep` component is a `react-native` only enhancer component, which is used to wrap the element that needs to be highlighted with a specific unique identifier.
+
+Another thing to note is that, the wrapped component needs to expose it's `ref` so that the `TourStep` component can collect the `ref` and save it to the state inside the `Tour` component.
+
+The usage example will look something like this:
+
+```jsx
+// ... Tour related components (their API stay mostly the same)
+
+// In some other file: DashboardPage.tsx
+const DashboardPage = () => {
+  return (
+    <Box>
+      <TourStep id="step-1">
+        <Button>Click me</Button>
+      </TourStep>
+      <Box>
+        <TourStep id="step-2">
+          <Box padding="spacing.5">
+            <Text>Some content</Text>
+          </Box>
+        </TourStep>
+      </Box>
+
+      <Card>
+        <CardBody>
+          <TourStep id="step-3">
+            <Box id="step-3">
+              <Text>Some content</Text>
+              <Text>Some content</Text>
+              <Text>Some content</Text>
+            </Box>
+          </TourStep>
+        </CardBody>
+      </Card>
+    </Box>
+  );
+};
+```
+
+This component is needed because in react-native there is no `id` prop or `getElementById` platform API like web which can be used to simply query the element and highlight it.
+
+Now the open question is:
+
+1. Should we have this API difference between web & native?
+
+**A few points to consider:**
+
+- This is absolutely needed on `react-native`
+- Technically we can also expose this component for `web` too and make the API surface same, although it will largely be useless because on web we can simply use the `id` prop to query the element and highlight it.
+
+The `web` implementation will look like this:
+
+```jsx
+const TourStep = ({ id, children }) => {
```

Comment:
chore: when using ref, are we planning to merge the ref from TourStep and component prop? might be useful for cases where ref is used on underlying children.

Reply:
Yes, we will ensure that consumer's ref doesn't get overriden. We will merge consumers ref and our internal ref together.